Release workflow fails to pack multi-target build artifacts for net8.0 and net9.0

## Description

Running the Release workflow via `workflow_dispatch` fails during the `dotnet pack` step. MSBuild cannot locate the compiled artifacts for `net8.0` and `net9.0`, even though the preceding `dotnet build` step reports success.

## Steps to Reproduce

1. Trigger the Release workflow manually via GitHub Actions (`workflow_dispatch`)
2. Observe the `Pack` step fail

## Expected Behaviour

`dotnet pack` successfully bundles all three target frameworks (`net8.0`, `net9.0`, `net10.0`) into the `.nupkg`.

## Actual Behaviour

The pack step fails with `MSB3030: Could not copy the file ... because it was not found` for `net8.0` and `net9.0` artifacts:

```
error MSB3030: Could not copy the file ".../bin/Release/net8.0/CodeToNeo4j.runtimeconfig.json" because it was not found. [CodeToNeo4j.csproj::TargetFramework=net8.0]
error MSB3030: Could not copy the file "obj/Release/net8.0/CodeToNeo4j.dll" because it was not found. [CodeToNeo4j.csproj::TargetFramework=net8.0]
error MSB3030: Could not copy the file "obj/Release/net8.0/CodeToNeo4j.pdb" because it was not found. [CodeToNeo4j.csproj::TargetFramework=net8.0]
error MSB3030: Could not copy the file "obj/Release/net9.0/CodeToNeo4j.dll" because it was not found. [CodeToNeo4j.csproj::TargetFramework=net9.0]
error MSB3030: Could not copy the file ".../bin/Release/net9.0/CodeToNeo4j.runtimeconfig.json" because it was not found. [CodeToNeo4j.csproj::TargetFramework=net9.0]
error MSB3030: Could not copy the file "obj/Release/net9.0/CodeToNeo4j.pdb" because it was not found. [CodeToNeo4j.csproj::TargetFramework=net9.0]
```

## Suspected Cause

The release workflow uses `clean: false` on checkout and `--no-build` on `dotnet pack`. It's possible that:

- The `dotnet build` step only produces `net10.0` artifacts (the active SDK), leaving `net8.0` / `net9.0` `obj/` outputs absent or stale
- `dotnet pack --no-build` then cannot locate the expected multi-TFM artifacts

## dotnet --list-sdks output

<!-- Please paste the output of `dotnet --list-sdks` from the self-hosted runner -->

## Environment

- Runner: self-hosted (macOS)
- Workflow: `.github/workflows/release.yml`
- Project targets: `net10.0;net9.0;net8.0`
- Active SDK from `global.json`: 10.0.103

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -37,7 +37,7 @@ jobs:
       run: dotnet test --no-build -c Release --verbosity normal
 
     - name: Pack
-      run: dotnet pack src/CodeToNeo4j/CodeToNeo4j.csproj --no-build -c Release -o out /p:Version=${{ env.MAJOR }}.${{ env.MINOR }}.${{ github.run_number }}
+      run: dotnet pack src/CodeToNeo4j/CodeToNeo4j.csproj --no-restore -c Release -o out /p:Version=${{ env.MAJOR }}.${{ env.MINOR }}.${{ github.run_number }}
 
     - name: Attest build provenance
       uses: actions/attest-build-provenance@v2

